### PR TITLE
[Feature 2] User needs to accept terms and conditions before proceeding to onboarding

### PR DESCRIPTION
**Pull request by alexxispn**
## Description

Install classnames library, implement Roboto Flex Font, rename components names and add new page: RegisterLegal.

## Changes

- Install classnames library and update AppNextButton.
- Implement Font Roboto Flex and create all styles that are used on Figma.
- Implement new font styles on all app components.
- Rename some components to have more coherence with the rest of the app.
- Order all tailwind classes on components for better coherence.

## Checks

- [X] Project Builds
- [X] Project passes tests and checks
- [x] Updated documentation accordingly

## Additional information

Reference: [2] User needs to accept terms and conditions before proceeding to onboarding
URL: https://github.com/TheTributeCommunity/fesbal-frontend/issues/3

This branch was created from the branch 'feature-1/build-initial-screen-layout' in order not to have to redo tasks already done.
URL: https://github.com/TheTributeCommunity/fesbal-frontend/pull/51

### CODE REVIEW

**Review by yodra — COMMENTED** [review]
I think there are other PRs with the same changes in these cases it's better to change the base branch to the principal branch to have in the PR just the changes of the current PR.

For example, If you create the first PR (branch_1), and then you need to create another (branch_2) with the branch_1 code, you can create the PR branch_2 to merge it in branch_1.

File: src/pages/RegisterID.tsx
Comment (albertopolofer) [reply src/pages/RegisterID.tsx:6]: +1, its an awesome job @alexxispn but it's really difficult to check the changes only related to this PR :)

**Comment by albertopolofer** [comment]
Couple of comments on minor issues i've found:

1. After clicking the register button it takes the user to the DNI/NIE registration, according to the wireframe it was supposed to be the user name page instead, being the last one the DNI page.

2. There is a glitch in the fields to introduce the DNI both the combo and he textfield. When you enter any value the field name goes to the upper position, and the text field itself moves too, disaligning with the other one, I think it would be nicer if it didn't move:
image

3. I've also seen problems with the responsiveness of the design, some screens can look a bit weird on some devices. We should make sure not that this looks great on any device but at least that whenever we address the design and style parts it can be done as easy as possible, this is where avoiding as much as possible inline css is paramount.
So, if it's just pending work to be done in css this would be minor issue, but otherwise we should try to address this before it snowballs.

Some examples:

Galaxy s20 ultra:
image

Galaxy s8+
image

Iphone 12 pro:
image

On tablet sized devices it can look weirder too.

File: src/components/atom/AppBurgerMenuButton.tsx
Comment (albertopolofer) [inline src/components/atom/AppBurgerMenuButton.tsx:5]: It's probably work in progress but I'd love to have a global style for app buttons so in code we will have something like "button-background" or "button-background burguer" or whatever.

File: src/types/ButtonProps.ts
Comment (albertopolofer) [inline src/types/ButtonProps.ts:4]: Not sure about this solution, looks like it's coupling code and styling, I think style should go always just in css and code or business related things should go in the ts/tsx part, makes sense to have title, enabling and links in the button properties but maybe it's better to have styling separated.

File: src/styles/variables.scss
Comment (albertopolofer) [inline src/styles/variables.scss:5]: 💯 love this styling names and great spirit on addressing typos, they tend to be there forever XD

**Review by albertopolofer — APPROVED** [review]
LGTM once reviewed comments

Please take a look at the comments before merging, I let the PR approved as they are mostly minor issues.

great job!

**Comment by alexxispn** [comment]
Hi @yodra  and @albertopolofer , I have addressed the majority of the comments in this pull request review. However, I am facing a difficulty in handling the implementation of the AppNextButton component. Specifically, I am unsure of how to remove the bgColor property while still being able to use the component with different background colors.

P.S. Also, the registration button currently redirects to the registrationID page as the registrationUser page is not yet implemented."

File: src/components/icons/LogoFesbalIcon.tsx
Comment (yodra) [inline src/components/icons/LogoFesbalIcon.tsx:3]: Check the icon it's showing correctly because you didn't update the viewBox proportion and it's possible the icon doesn't respect the aspect ratio

**Review by yodra — APPROVED** [review]

**Pull request merged**
